src/main/scala/mimir/algebra/function/IntervalFunctions.scala

```diff
@@ -0,0 +1,90 @@
+package mimir.algebra.function;
+
+import mimir.algebra._
+import org.joda.time.DateTime
+
+object IntervalFunctions
+{
+
+  def register()
+  {
+    FunctionRegistry.registerNative("TOMILLISECONDS", 
+      (x) => FloatPrimitive(x(0).asLong), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("TOSECONDS", 
+      (x) => FloatPrimitive(x(0).asLong / 1000), 
+      (_) => TInt()
+    )
+
+
+    FunctionRegistry.registerNative("TOMINUTES", 
+      (x) => FloatPrimitive(x(0).asLong /1000 / 60), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("TOHOURS", 
+      (x) => FloatPrimitive(x(0).asLong /1000 / 60 / 60), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("TODAYS", 
+      (x) => FloatPrimitive(x(0).asLong /1000 / 60 / 60 / 24), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("PLUSMILLISECONDS", 
+      (x) => IntervalPrimitive(x(0).asLong + x(1).asLong), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("PLUSSECONDS", 
+      (x) => IntervalPrimitive(x(0).asLong + x(1).asLong * 1000), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("PLUSMINUTES", 
+      (x) => IntervalPrimitive(x(0).asLong + x(1).asLong * 1000 * 60), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("PLUSHOURS", 
+      (x) => IntervalPrimitive(x(0).asLong + x(1).asLong * 1000 * 60 * 60), 
+      (_) => TInt()
+    )
+
+    FunctionRegistry.registerNative("PLUSDAYS", 
+      (x) => IntervalPrimitive(x(0).asLong + x(1).asLong * 1000 * 60 * 60 * 24), 
+      (_) => TInt()
+    )
```

Comment:
These manual addition operations aren't really needed. Basic arithmetic operations should take care of it (though see my comment on units and `IntervalPrimitive` above).  You should have an `INTERVAL` function capable of parsing intervals from strings (essentially a CAST operation).  JODA time's Period object has a [parse method](http://www.joda.org/joda-time/apidocs/org/joda/time/Period.html#parse-java.lang.String-) for this, though I'd do a quick sanity check first to see if it matches up with [SQL Interval Semantics](http://www.techrepublic.com/article/sql-basics-datetime-and-interval-data-types/).